node/src/blockchain/blockchain_interface.rs

```diff
@@ -275,40 +307,51 @@ where
                             let transactions: Vec<BlockchainTransaction> = logs
```

Comment:
This hunk of code seems like something that could live in an outboard method and make the caller clearer.